### ISSUE TITLE
Store setup failed / SMW 2.5 support

### ISSUE THREAD

**Issue by kghbln** [rdfio/RDFIO#33]
**Setup**
- MediaWiki 	1.28.0 (fec3e69) 18:52, 1 March 2017
- PHP 	5.6.30-0+deb8u1 (apache2handler)
- MariaDB 	10.0.30-MariaDB-1~jessie
- Semantic MediaWiki	2.5.0 (85b2386) 18:36, 25 March 2017
- RDFIO	v2.0.9 (cd0e3c6) 04:02, 27 November 2016

**Issue**
It does not seem possible to create the required database tables while running MediaWiki's `update.php` maintenance script. So I went to special page "RDFIOAdmin" and hit the "septup" button. Then I got the "Setup failed:" note however without a message. The error and exception logs are empty.

**Comment by samuell** [comment]
Oops, Thanks so much for reporting @kghbln !

I guess it is something that happened after SMW 2.5 ... (Tested only 2.4 so far).

Will have a look!

**Comment by kghbln** [comment]
Thanks a lot. As the matter of fact the wiki is running 3.0.0-alpha of SMW. Due to caching special page "Version" did not show the latest version. Talking about this special page. It seems that every special page of RDFIO registers itself so things appear a bit cluttered. Probably you may want to change this too.

Once fluffed I would like to install this extension on <sandbox.smw.o> which runs master. Thus issue can be detected earlier.

**Comment by samuell** [comment]
Ah, yea, I see what you mean about the special pages! I think I followed some guide back in time, and though this was how to do it, although I realize now the guide probably focused on creating a complete extension, around a special page! Thanks for the heads up! :+1: :)

**Comment by kghbln** [comment]
> I realize now the guide probably focused on creating a complete extension, around a special page! 

I guess so. No worries. There is always room to improve. :) I really appreciate your effort and that's why I think it should also be served on <sandbox.smw.o>!

**Comment by samuell** [comment]
> I guess so. No worries. There is always room to improve. :) I really appreciate your effort and that's why I think it should also be served on <sandbox.smw.o>!

Thank you, that would be cool!

**Comment by samuell** [comment]
@kghbln Sorry for my ignorance, bit curios question ... what is the meaning of the id "sandbox.smw.o"? Is it a domain name? (Could not access it). :)

**Comment by kghbln** [comment]
> I gather it is the same as https://sandbox.semantic-mediawiki.org/wiki/Main_Page right?

Yep, it is this one. Actually we always had this sandbox but it was not maintained and defunct for about half a year until I started to serve it a year and so ago. If I write the full domain name in brackets then GitHub muches the link.

**Comment by samuell** [comment]
After some experimentation, it turns out it might be that ARC2 is not getting properly initialized with the DB credentials (user name and password). 

I strangely seem to be able to install ARC2 tables in the vagrant box, but upon trying to import the test data, I get:

```
[vagrant w]$ php maintenance/importDump.php /vagrant/roles/rdfio/files/wiki_content.xml
Store config: PHP Warning:  mysqli_connect(): (28000/1045): Access denied for user 'vagrant'@'localhost' (using password: NO) in /var/www/html/w/vendor/semsol/arc2/store/ARC2_Store.php on line 51
```

Probably related to [the hack needed to access info from LocalSettings.php](https://github.com/rdfio/RDFIO/issues/13#issuecomment-256414481).

**Comment by samuell** [comment]
Obviously, SMW 2.5 is installed, with the composer command:

```
composer require mediawiki/semantic-media-wiki ~2.4 --update-no-dev
```

... don't understand why ...

**Comment by kghbln** [comment]
> ... don't understand why ...

We are not lost in translation but lost in dependency I'm afraid. I am getting 2.5.x too though I would like to have 3.0.x https://github.com/SemanticMediaWiki/semantic-mediawiki.org/issues/27

**Comment by samuell** [comment]
>> ... don't understand why ...
>
> We are not lost in translation but lost in dependency I'm afraid. I am getting 2.5.x too though I would like to have 3.0.x https://github.com/SemanticMediaWiki/semantic-mediawiki.org/issues/27

Heh, yes! :) Though, now I got some great advice by @mwjames on the mailing list, so this might resolve things for me at least :)

**Comment by mwjames** [comment]
```
[vagrant maintenance]$ php importDump.php /vagrant/roles/rdfio/files/wiki_content.xml
Store config: [16a44ce8] [no req]   RDFIOARC2StoreException from line 138 of /var/www/html/w/extensions/Rdfio/stores/SMW_ARC2Store.php: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1
Backtrace:
#0 /var/www/html/w/extensions/Rdfio/stores/SMW_ARC2Store.php(162): SMWARC2Store->executeArc2Query(string)
#1 /var/www/html/w/extensions/Rdfio/stores/SMW_ARC2Store.php(47): SMWARC2Store->removeDataForURI(string)
#2 /var/www/html/w/extensions/SemanticMediaWiki/includes/StoreUpdater.php(218): SMWARC2Store->updateData(SMW\SemanticData)
```

I'm not sure why "SMW_ARC2Store.php: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1" is triggered or returned but this error is emitted by `SMWARC2Store` and not SMW-core.

The code doesn't reveal much only that something goes wrong during `$this->arc2store->query( $q )`.

```
        $q = $requestString;
        $rs = $this->arc2store->query( $q );
        $result = $rs;
        $errors = $this->arc2store->getErrors();
        foreach ( $errors as $error ) {
        	throw new RDFIOARC2StoreException( $error );
        }
```
```
 $this->arc2store = ARC2::getStore( $smwgARC2StoreConfig );
```

**Comment by samuell** [comment]
Now, with MW 1.29, SMW 2.5.4 and RDFIO 2.3.1, I get [this](https://gist.github.com/samuell/625f5408c76cbc1aab23f6e202cc3daf).

**Comment by samuell** [comment]
Turns out the method of accessing user info variables from LocalSettings.php, [here](https://github.com/rdfio/RDFIO/blob/711762f51491c9523b0440db057649654379d114/RDFIO.php#L71-L91) does not work any more.

**Comment by samuell** [comment]
Fix now well under way, with 8ab2507d23da45ee4f7c1af3ba1f3a9490f72c4b ! (Just more testing needed)

**Comment by samuell** [comment]
Fixed now, in [v2.4.0](https://github.com/rdfio/RDFIO/releases/tag/v2.4.0).
